Makefile

```diff
@@ -30,7 +30,7 @@ PYFORMAT=$(TOOLS)/pyformat
 RSTLINT=$(TOOLS)/rst-lint
 
 BROKEN_VIRTUALENV=$(BUILD_RUNTIMES)/virtualenvs/broken
-TOOL_VIRTUALENV=$(BUILD_RUNTIMES)/virtualenvs/tools
+TOOL_VIRTUALENV:=$(BUILD_RUNTIMES)/virtualenvs/tools-$(shell sha1sum requirements/tools.txt | head -c 10)
```

Comment:
I don’t know how much you care, but this only works on Linux – neither macOS nor Windows have the `sha1sum` command. On those platforms, you see `/bin/sh: sha1sum: command not found` in the output and get a virtualenv named `tools-`.

AFAIK, the only cross-platform way to compute SHA1 sums is to drop into Python (ick) – shell utilities or OpenSSL are inconsistent.